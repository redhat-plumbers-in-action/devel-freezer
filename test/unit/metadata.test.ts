import { describe, it, expect, beforeEach, test } from 'vitest';

import {
  metadataContextFixture,
  IMetadataTestContext,
} from './fixtures/metadata.fixture';

describe('Metadata Object', () => {
  beforeEach<IMetadataTestContext>(context => {
    context.metadata = metadataContextFixture.metadata;
    context.invalid = metadataContextFixture.invalid;
  });

  it<IMetadataTestContext>('can be instantiated', context =>
    context.metadata.map(metadataItem => expect(metadataItem).toBeDefined()));

  it.todo<IMetadataTestContext>(
    'is invalid' /*, async context => {
    context.invalid.map(async item =>
      expect(Metadata.validate(item)).resolves.toMatchSnapshot()
    );
  }*/
  );

  test<IMetadataTestContext>('get tag()', context =>
    context.metadata.map(metadataItem =>
      expect(metadataItem.tag).toMatchSnapshot()
    ));

  test.todo<IMetadataTestContext>('set tag()');

  test<IMetadataTestContext>('get commentID()', context =>
    context.metadata.map(metadataItem =>
      expect(metadataItem.commentID).toMatchSnapshot()
    ));

  test.todo<IMetadataTestContext>('set commentID()');

  test.todo<IMetadataTestContext>('getMetadata()');
  test.todo<IMetadataTestContext>('setMetadata()');
});
