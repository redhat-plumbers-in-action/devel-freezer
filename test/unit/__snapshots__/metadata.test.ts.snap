// Vitest Snapshot v1

exports[`Metadata Object > get commentID() 1`] = `undefined`;

exports[`Metadata Object > get commentID() 2`] = `undefined`;

exports[`Metadata Object > get commentID() 3`] = `"123456789"`;

exports[`Metadata Object > get commentID() 4`] = `"123456789"`;

exports[`Metadata Object > get tag() 1`] = `undefined`;

exports[`Metadata Object > get tag() 2`] = `"v1"`;

exports[`Metadata Object > get tag() 3`] = `undefined`;

exports[`Metadata Object > get tag() 4`] = `"v1"`;
